test/integration/fiscalYear.extra.js

```diff
@@ -0,0 +1,144 @@
+         * FIXME: timezone issue : we have 2017-12-31 in the database
+         * but we get 2018-01-01, it is now difficult to test date.
+         */
+        expect(fy).to.have.property('start_date');
+        expect(fy).to.have.property('end_date');
+        const fyStartDate = flatDate(fy.start_date);
+        const fyEndDate = flatDate(fy.end_date);
+        expect(fyStartDate).to.be.equal('2017-01-01');
+        expect(fyEndDate).to.be.equal('2017-12-31');
+
+        /**
+         * the percentage is value of consumed period in the fiscal year
+         * until to the given date
+         * proportion = (currentDate - startDate) / (nbMonthOfYear * 30.5)
+         * proportion = ('2017-08-24' - '2017-01-01') / (12 * 30.5)
+         * proportion = 235 / 366 = 0.6421 = 64.21%
+         */
+        expect(fy).to.have.property('percentage', 0.6421);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('POST /fiscal/:id/opening_balance forbid to set the opening balance of a fiscal year which is not the first', () => {
+    // define a not first fiscal year information
+    openingBalance.id = fiscalYear2017.id;
+    openingBalance.fiscal = fiscalYear2017;
+    openingBalance.accounts = accounts;
+
+    return agent.post(url.concat(`/${fiscalYear2017.id}/opening_balance`))
+      .send({ params : openingBalance })
+      .then(res => {
+        helpers.api.errored(res, 400, 'ERRORS.NOT_BEGINING_FISCAL_YEAR');
+      })
+      .catch(helpers.handler);
+  });
+
+  it('POST /fiscal/:id/opening_balance set the opening balance of a fiscal year', () => {
+    // define a not first fiscal year information
+    openingBalance.id = fiscalYear2015.id;
+    openingBalance.fiscal = fiscalYear2015;
+    openingBalance.accounts = accounts;
+
+    return agent.post(url.concat(`/${fiscalYear2015.id}/opening_balance`))
+      .send({ params : openingBalance })
+      .then(res => {
+        expect(res).to.have.status(201);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /fiscal/:id/opening_balance returns the opening balance of a given fiscal year', () => {
+    return agent.get(url.concat(`/${fiscalYear2015.id}/opening_balance`))
+      .then(res => {
+        const updatedAccounts = res.body.filter(item => {
+          return [81, 83, 174, 190].indexOf(item.id) > -1;
+        });
+        const [first, second, third, fourth] = updatedAccounts;
+        expect(first.balance).to.be.equal(77);
+        expect(second.balance).to.be.equal(-77);
+        expect(third.balance).to.be.equal(23);
+        expect(fourth.balance).to.be.equal(-23);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /fiscal/:id/periods returns periods of the given fiscal year', () => {
+    return agent.get(url.concat(`/${fiscalYear2017.id}/periods`))
+      .then(res => {
+        expect(res.body).to.have.length(fiscalYear2017MonthsNumber + 1);
+        for (let i = 0; i < res.body.length; i++) {
+          expect(res.body[i].number).to.be.equal(i);
```

Comment:
This took me a minute to understand what it was doing.  Could you leave a comment?  It seems to me it is checking that the fiscal year periods are returned in order.